### DISCUSSION TITLE
Supabase on ARM

### DISCUSSION THREAD

**Discussion by mrvillage** [supabase/supabase#5328] [Questions]
Hello!

I've been trying to get Supabase to run on ARM and have been running into some issues. The containers docker-compose is pulling from Docker Hub are built for ARM. However half of them fail to start (supabase-storage, supabase-auth, supabase-meta, and supabase-db) all printing the same error `standard_init_linux.go:228: exec user process caused: exec format error`.

Does anyone have any potential solutions? Thanks in advance!

**Comment by inian** [comment]
Are you running the latest version of the CLI @mrvillage? At least for storage, we have just started publishing ARM images to docker hub too which the latest version of our CLI [is using](https://github.com/supabase/cli/commit/58f77591a6b6f2962616521ad2121f43772d1589).

**Reply by mrvillage** [reply]
Thanks you so much for the response @inian! Sorry for the delay, I'm not using the CLI, I was following the instructions on the self-hosting page to install via docker. I updated my docker-compose to use v0.11.0 of the storage-api and that worked like a charm, updating supabase-auth to the latest release also fixed that container. However, the supabase-meta and supabase-db instances are still restart looping with the same error. My theory is that the actual software in the container was built for amd64 not arm64, but I don't have enough experience to know for certain.

**Reply by mrvillage** [reply]
Update: supabase-meta now works on v0.33.2 instead of v0.29.0, however supabase-db is still not functional.

**Reply by mrvillage** [reply]
Another update, after cloning the supabase/postgres repo and building it myself with the $PLATFORM environment variable set to linux/arm64 as opposed to the linux/amd64 the image on Docker Hub was built with. Thank you for all your help!

**Reply by jtkeyva** [reply]
@mrvillage are you able to run this on M1? Can you provide the steps required to make this work?
Thanks

**Comment by jtkeyva** [comment]
@mrvillage I eventually got it working simply by increasing the RAM in Docker. I cranked it up to 16gb and now it's working on my M1 MBP. Hopefully a non emulated version will release soon along with Storage :)

**Reply by mrvillage** [reply]
Awesome! The only way I got it to work was building it for ARM manually, I believe all I had to do was `docker build --platform=linux/arm64`, I also built it on an ARM VM so that might have helped as well.

**Reply by jtkeyva** [reply]
Thank you!

**Comment by hub-unibloom** [comment]
Does it currently run everything smoothly? I'm looking to use a free Oracle instance and I wanted to know if the ARM issue was resolved properly or if there are still problems.

Ampere ARM-based processor. 
VM.Standard.A1.Flex Always Free-eligible 1 (80 max) 6 (512 max)
